lib/swiss_db/swiss_model.rb

```diff
@@ -1,121 +1,140 @@
-  end
+    def update_attributes(hash)
+      hash.each { |k, v| update_attribute(k, v) }
+    end
 
-  def update_attributes(hash)
-    hash.each do |k, v|
-      update_attribute(k, v)
+    def primary_key
+      self.class.primary_key
     end
-  end
 
-  def self.store
-    context = DataStore.context
-    @store ||= DataStore.new(context)
-    @store
-  end
+    def primary_key_value
+      self.send(primary_key.to_sym)
+    end
 
-  def self.class_name
-    @class_name
-  end
+    def table_name
+      self.class.table_name
+    end
 
-  def self.set_class_name(class_name) # hack, class.name not functioning in RM Android...
-    @class_name = class_name
-    set_table_name(class_name.tableize)
-  end
+    def store
+      self.class.store
+    end
 
-  def self.set_table_name(table_name)
-    @table_name = table_name
-  end
+    # def destroy
+    #   # destroy this row
+    # end
 
-  def self.table_name
-    @table_name
-  end
+    # -------------
+    # CLASS METHODS
+    # -------------
+    class << self
+      def store
+        SwissDB.store
+      end
 
-  def self.set_primary_key(primary_key)
-    @primary_key = primary_key
-  end
+      def class_name
+        @class_name
+      end
 
-  def self.primary_key
-    @primary_key.nil? ? "id" : @primary_key
-  end
+      def set_class_name(class_name) # hack, class.name not functioning in RM Android...
+        @class_name = class_name
+        set_table_name(class_name.tableize)
+      end
 
-  def self.all
-    # select_all
-    cursor = store.select_all(@table_name, self)
-    cursor
-  end
+      def set_table_name(table_name)
+        @table_name = table_name
+      end
 
-  def self.where(values)
-    # select <table> where <field> = <value>
-    cursor = store.select(@table_name, values, self)
-    cursor
-  end
+      def table_name
+        @table_name
+      end
 
-  def self.first
-    # select all and get first
-    cursor = all.first
-    cursor
-  end
+      def set_primary_key(primary_key)
+        @primary_key = primary_key
+      end
 
-  def self.last
-    # select all and get last
-    cursor = all.last
-    cursor
-  end
+      def primary_key
+        @primary_key.nil? ? "id" : @primary_key
+      end
 
-  def self.create(obj)
-    # create a row
-    result = store.insert(@table_name, obj)
-      if result == -1
-        puts "An error occured inserting values into #{@table_name}"
-      else
-        return result
+      def all
+        # select_all
+        cursor = store.select_all(@table_name, self)
+        cursor
       end
-  end
 
-  # def destroy
-  #   # destroy this row
-  # end
+      def where(values)
+        # select <table> where <field> = <value>
+        cursor = store.select(@table_name, values, self)
+        cursor
+      end
 
-  def self.destroy_all!
-    # destroy all of this kind (empty table)
-    store.destroy_all(@table_name)
-  end
+      def first
+        # select all and get first
+        cursor = all.first
+        cursor
```

Comment:
this should be called `model` not cursor. `model#all` does return a cursor but `all.first` returns a new instance of the model.